vernac/record.ml

```diff
@@ -59,26 +59,37 @@ let () =
       optread  = (fun () -> !typeclasses_unique);
       optwrite = (fun b -> typeclasses_unique := b); }
 
-let interp_fields_evars env sigma impls_env nots l =
-  List.fold_left2
-    (fun (env, sigma, uimpls, params, impls) no ({CAst.loc;v=i}, b, t) ->
-      let sigma, (t', impl) = interp_type_evars_impls ~program_mode:false env sigma ~impls t in
-      let r = Retyping.relevance_of_type env sigma t' in
-      let sigma, b' =
-        Option.cata (fun x -> on_snd (fun x -> Some (fst x)) @@
-                      interp_casted_constr_evars_impls ~program_mode:false env sigma ~impls x t') (sigma,None) b in
-      let impls =
-        match i with
-        | Anonymous -> impls
-        | Name id -> Id.Map.add id (compute_internalization_data env sigma Constrintern.Method t' impl) impls
-      in
-      let d = match b' with
-              | None -> LocalAssum (make_annot i r,t')
-              | Some b' -> LocalDef (make_annot i r,b',t')
+let interp_fields_evars env sigma ~ninds ~nparams impls_env nots l =
+  let _, sigma, impls, newfs, _ =
+    List.fold_left2
+      (fun (env, sigma, uimpls, params, impls) no ({CAst.loc;v=i}, b, t) ->
+         let sigma, (t', impl) = interp_type_evars_impls env sigma ~impls t in
+         let r = Retyping.relevance_of_type env sigma t' in
+         let sigma, b' =
+           Option.cata (fun x -> on_snd (fun x -> Some (fst x)) @@
+                         interp_casted_constr_evars_impls ~program_mode:false env sigma ~impls x t') (sigma,None) b in
+         let impls =
+           match i with
+           | Anonymous -> impls
+           | Name id -> Id.Map.add id (compute_internalization_data env sigma Constrintern.Method t' impl) impls
+         in
+         let d = match b' with
+           | None -> LocalAssum (make_annot i r,t')
+           | Some b' -> LocalDef (make_annot i r,b',t')
+         in
+         List.iter (Metasyntax.set_notation_for_interpretation env impls) no;
+         (EConstr.push_rel d env, sigma, impl :: uimpls, d::params, impls))
+      (env, sigma, [], [], impls_env) nots l
+  in
+  let _, sigma = Context.Rel.fold_outside ~init:(env,sigma) (fun f (env,sigma) ->
+      let sigma = RelDecl.fold_constr (fun c sigma ->
+          ComInductive.maybe_unify_params_in env sigma ~ninds ~nparams c)
```

Comment:
Adding this (+ exposing the function from cominductive) solves the record issue.